src/safe-unsafe-meaning.md

```diff
@@ -123,23 +125,31 @@ Rust has traditionally avoided doing this because it makes Unsafe
 Rust pervasive, which isn't desirable. `Send` and `Sync` are marked unsafe
 because thread safety is a *fundamental property* that unsafe code can't
 possibly hope to defend against in the way it could defend against a bad
-`Ord` implementation. The decision of whether to mark your own traits `unsafe`
-depends on the same sort of consideration. If `unsafe` code can't reasonably
-expect to defend against a bad implementation of the trait, then marking the
-trait `unsafe` is a reasonable choice.
+`Ord` implementation. Similarly, `GlobalAllocator` is keeping accounts of all
+the memory in the program and nobody can hope to deal with things like the
+allocator serving two different requests with the same chunk of memory. The
+decision of whether to mark your own traits `unsafe` depends on the same sort of
+consideration. If `unsafe` code can't reasonably expect to defend against a bad
+implementation of the trait, then marking the trait `unsafe` is a reasonable
```

Comment:
Might be worth expanding on "bad". (I know the original text didn't do that, so feel free to ignore this comment if you prefer.) Concretely, the point is that we expect the trait to follow more rules than the types suggest (this applies to `Ord` and `GlobalAllocator` alike), and that unsafe code can *rely* on the trait following these rules. The latter part is mentioned but I feel it is worth saying that there is a base promise made for every trait ("the methods act safety at their given type"), and `unsafe` merely says that we require more.